sqlmesh/core/engine_adapter/bigquery.py

```diff
@@ -87,6 +87,18 @@ def _job_params(self) -> t.Dict[str, t.Any]:
             params["maximum_bytes_billed"] = self._extra_config.get("maximum_bytes_billed")
         return params
 
+    def _begin_session(self) -> None:
+        from google.cloud.bigquery import QueryJobConfig
+
+        job = self.client.query("SELECT 1;", job_config=QueryJobConfig(create_session=True))
```

Comment:
Ah so you can't actually just pass in a session ID then? I was thinking you had to get it from BQ.

Reply:
yep, you were right after all.

Reply:
Cool making sure we still explored passing it in. That would have been nice. 😃